addons/survey/static/src/views/widgets/set_flag_fields/set_flag_fields.js

```diff
@@ -0,0 +1,84 @@
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { booleanField, BooleanField } from "@web/views/fields/boolean/boolean_field";
+import { integerField, IntegerField } from "@web/views/fields/integer/integer_field";
+
+import { useEffect, useRef } from "@odoo/owl";
+
+export class IntegerSetFlag extends IntegerField {
```

Comment:
Wait what ?

Reply:
Maybe ``IntegerUpdateFlag"`` maybe ? Better hint at the fact it flags "real updates". I would also add a docstring, because it would be cool to have a class docstring explaining quickly what it does, why we use it and how to use it. Here people have to retro engineer the code and guess why we did it.